app/views/admin/products_v3/_filters.html.haml

```diff
@@ -1,6 +1,6 @@
-= form_with url: admin_products_path, id: "filters", method: :get, data: { "search-target": "form", 'turbo-frame': "_self" } do
+= form_with url: admin_products_path, id: "filters", method: :get, data: { "search-target": "form", 'turbo-frame': "_self", 'turbo-action': "advance" } do
```

Comment:
Hi @dacook - Here are my findings on the issue raised by the @filipefurtad0:
- Due to `'turbo-frame': "_self"` on the above form submission, it targets the following frame, hence it gets the `aria-busy="true"` attribute added when it's being updated.
https://github.com/openfoodfoundation/openfoodnetwork/blob/5caeb160efe36906ddb17a0373860358b60cef6f/app/views/admin/products_v3/_content.html.haml#L1
- This causes the loader to show up as per our CSS rules
- The root cause here is that this same page (with `aria-busy="true"`) is then cached by Turbo, and upon navigation back, the `aria-busy="true"`  is still present on the (cached) page
- I think this is a bug in Turbo, what do you think? It should sanitize the page so that no `aria-busy` is present on any frame. Or cache the page before making it busy. 

For the solution, I had many, For example:
- remove `'turbo-frame': "_self", 'turbo-action': "advance"` from the affected links so that by our default rule, each page will be replaced by "_top" frame
- This fixes our current loader problem, however, by this approach, we won't be getting the `aria-busy="true"` on the above-mentioned frame, so no loader upon UI button clicks.

Another solution is that I think we should opt for this one:
- We should disable the Turbo cache because by default, we are also not caching pages in the browser, hence navigation back gets the page by the server.
![image](https://github.com/user-attachments/assets/7de963a5-f1b5-4976-acaf-31a6166a392d)

- I've implemented this here, please review: https://github.com/openfoodfoundation/openfoodnetwork/pull/12678/commits/a6efad73a8802634ce3610deee5c57e6fb4279cc

Please let me know what you think about it, if you are okay with this, we are good to move it back to testing. Thanks

Reply:
Hi Ahmed, this looks tricky. I'm sorry I'm not feeling well today and am not able to fully understand. A few thoughts:

I'm wondering what difference it makes if we remove the `morph` method?

I'm hesitant to disable cache entirely in case it would be helpful in other circumstances. I'm wondering if choosing `no-preview` would help (https://turbo.hotwired.dev/handbook/building#opting-out-of-caching)?
But If `no-cache` is the only way to solve it, then go ahead.

Reply:
> Hi Ahmed, this looks tricky. I'm sorry I'm not feeling well today and am not able to fully understand. A few thoughts:
> 
> I'm wondering what difference it makes if we remove the `morph` method?
> 
> I'm hesitant to disable cache entirely in case it would be helpful in other circumstances. I'm wondering if choosing `no-preview` would help (https://turbo.hotwired.dev/handbook/building#opting-out-of-caching)? But If `no-cache` is the only way to solve it, then go ahead.

No worries, David. Please take a rest and get well soon.

I've tried using `no-preview` as cache control and removing morph as well, however, it's of no use. The issue explained [here](https://github.com/openfoodfoundation/openfoodnetwork/pull/12678#discussion_r1686205511) remains the same.

In the context of Turbo, Morphing is a way to handle page reloads/refreshes. When a page/frame refresh is set to morph, then Turbo doesn't fully replace the page/frame. It calculates the changes between the current HTML response and the current page/frame and only updates the changes in the current frame/page, rather than the full replacement.
Ref: https://turbo.hotwired.dev/handbook/page_refreshes#morphing

I'm afraid, disabling Turbo cache seems to be the only solution.

Reply:
Thanks Ahmed, I guess that's the only option, please go ahead.